File: backend/src/main/kotlin/outcobra/server/service/FilterService.kt
package outcobra.server.service

import outcobra.server.model.dto.filter.SubjectFilterDto

/**
 * @author Florian Bürgi
 * @since 1.2.0
 */
interface FilterService {

    /**
     * @return a [SubjectFilterDto] for the current user
     */
    fun getSubjectFilter(): SubjectFilterDto
}

File: frontend/src/app/core/services/filter/filter.service.ts
import {AppService} from '../core/app.service';
import {HttpInterceptor} from '../../http/http-interceptor';
import {Injectable} from '@angular/core';
import {SubjectFilterDto} from '../../../task/model/subject-filter.dto';
import {Observable} from 'rxjs/Observable';

@Injectable()
export class FilterService extends AppService {
    constructor(http: HttpInterceptor) {
        super(http, "/filter");
    }

    public getSubjectFilter(): Observable<SubjectFilterDto> {
        return this._http.get<SubjectFilterDto>(`/subject${this._baseUri}`, 'outcobra');
    }
}
